src/superqt/utils/_code_syntax_highlight.py

```diff
@@ -1,81 +1,267 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, cast
+
 from pygments import highlight
 from pygments.formatter import Formatter
 from pygments.lexers import find_lexer_class, get_lexer_by_name
 from pygments.util import ClassNotFound
-from qtpy import QtGui
+from qtpy.QtGui import (
+    QColor,
+    QFont,
+    QPalette,
+    QSyntaxHighlighter,
+    QTextCharFormat,
+    QTextDocument,
+)
 
-# inspired by  https://github.com/Vector35/snippets/blob/master/QCodeEditor.py
-# (MIT license) and
-# https://pygments.org/docs/formatterdevelopment/#html-3-2-formatter
+if TYPE_CHECKING:
+    from collections.abc import Mapping, Sequence
+    from typing import Literal, TypeAlias, TypedDict, Unpack
+
+    import pygments.style
+    from pygments.style import _StyleDict
+    from pygments.token import _TokenType
+    from qtpy.QtCore import QObject
+
+    class SupportsDocumentAndPalette(QObject):
+        def document(self) -> QTextDocument | None: ...
+        def palette(self) -> QPalette: ...
+        def setPalette(self, palette: QPalette) -> None: ...
+
+    KnownStyle: TypeAlias = Literal[
```

Comment:
any plan for trace pigments style updates (ex compare pygments.styles.STYLES content witrh this annotation in tests) or just wait for issues?

Reply:
no plan... but I'll mention that the type hint here is actually:

```python
theme: KnownStyle | str = "default",
```

which means that users will get the autocomplete suggestions, but it's not an error to enter something that is not in the KnownStyle list.  So the only "issue" we might see is that someone didn't get a nice autocomplete hint :)